### PR TITLE
902 optimization pattern

### PR DESCRIPTION
**Pull request by lecourtoisflo**
Remove pattern using interface in Dynawo API when not necessary.

### CODE REVIEW

**Review by gautierbureau — APPROVED** [review]

File: dynawo/3rdParty/libzip/include/libzip/ZipEntry.h [hunk 1]
```diff
@@ -12,21 +12,27 @@
 #ifndef __ZIP_ENTRY_H__
 #define __ZIP_ENTRY_H__
 
+#include <archive_entry.h>
```
Comment (rosiereflo) [inline dynawo/3rdParty/libzip/include/libzip/ZipEntry.h:15]: Can't we split this huge MR into 1 MR per library? This woul be to limit the risk of a massive breakage

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Comment by rosiereflo** [comment]
What is the performance gain with this?

**Comment by lecourtoisflo** [comment]
> What is the performance gain with this?

Performances are unchanged ; duration are within the error margin

**Comment by rosiereflo** [comment]
Then we should discuss on the objective of this MR: I understood  that it was to improve performance so if we do not observe any performance gain I am not sure this should be merged as it is a big change that largely breaks code history

**Comment by lecourtoisflo** [comment]
> Then we should discuss on the objective of this MR: I understood that it was to improve performance so if we do not observe any performance gain I am not sure this should be merged as it is a big change that largely breaks code history

It was more for code clarification with a potential performance increase.

File: dynawo/sources/Solvers/Common/DYNSolverImpl.hpp [hunk 2]
```diff
@@ -11,7 +11,7 @@
 //
 
 /**
- * @file  DYNSolverImpl.hpp
+ * @file  DYNSolver.hpp
```
Comment (rosiereflo) [inline dynawo/sources/Solvers/Common/DYNSolverImpl.hpp:14]: not yet renamed :)

File: dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponentImpl.hpp [hunk 3]
```diff
@@ -11,7 +11,7 @@
 //
 
 /**
- * @file  DYNNetworkComponentImpl.hpp
+ * @file  DYNNetworkComponent.hpp
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNNetworkComponentImpl.hpp:14]: not yet renamed :)

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Comment by lecourtoisflo** [comment]
all NRT are OK

**Review by rosiereflo — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**